src/builtin.c

```diff
@@ -807,6 +806,62 @@ static jv f_sort_by_impl(jq_state *jq, jv input, jv keys) {
   }
 }
 
+/* Assuming the input array is sorted, bsearch/1 returns */
+/* the index of the target if the target is in the input array; and otherwise */
+/*  (-1 - ix), where ix is the insertion point that would leave the array sorted. */
+/* If the input is not sorted, bsearch will terminate but with irrelevant results. */
+static jv f_bsearch(jq_state *jq, jv input, jv target) {
+    assert(jv_get_kind(input) == JV_KIND_ARRAY);
+    assert(jv_get_kind(target) == JV_KIND_NUMBER);
+    int len = jv_array_length(jv_copy(input));
+    if (len == 0) {
+        return jv_number(-1);
+    } else if (len == 1) {
+        jv val = jv_copy(jv_array_get(jv_copy(input), 0));
+        int result = jv_cmp(target, val);
+        jv_free(val);
```

Comment:
useless copy+free?

Reply:
Yes.  Write this instead:

```C
        int result = jv_cmp(target, jv_copy(jv_array_get(jv_copy(input), 0)));
```

Reply:
Do we need to optimize for the case where `len == 1`?